subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c

```diff
@@ -816,14 +823,15 @@ static void isr_rx(void *param)
 
 	/* Next BIS */
 	if (lll->bis_curr < lll->num_bis) {
-		const uint8_t stream_curr = lll->stream_curr + 1U;
 		struct lll_sync_iso_stream *sync_stream;
 		uint16_t stream_handle;
 
 		/* Next selected stream */
-		if (stream_curr < lll->stream_count) {
+		stream_curr = lll->stream_curr + 1U;
+		if ((stream_curr < BT_CTLR_SYNC_ISO_STREAM_MAX) &&
+		    (stream_curr < lll->stream_count)) {
```

Comment:
Seems redundant to have both checks, as I assume that `stream_count` is never larger than `BT_CTLR_SYNC_ISO_STREAM_MAX`?

Reply:
See my previous reply, this is related to handling compiler warning.

Reply:
Aha. Interesting. 

Could it be, in both functions if needed, replaced by an ASSERT? 

If `stream_curr` ever gets larger than `BT_CTLR_SYNC_ISO_STREAM_MAX`, or rather if `lll->stream_count` ever gets larger than `BT_CTLR_SYNC_ISO_STREAM_MAX` that's a significant issue that shouldn't silent be ignored like this check seemingly does.

Reply:
`stream_count` is assigned after parameter validation in `ll_big_sync_create()`, the check here with `BT_CTLR_SYNC_ISO_STREAM_MAX`  is already redundant.

Reply:
> `stream_count` is assigned after parameter validation in `ll_big_sync_create()`, the check here with `BT_CTLR_SYNC_ISO_STREAM_MAX` is already redundant.

Do that mean you'll remove it, or what's the plan?

Reply:
This is the warning I get for bsim builds GCC 14.2.1 (not for target build using zephyr sdk), what do you suggest?

```
[121/198] Building C object zephyr/subsys/bluetooth/controller/ll_sw/nordic/CMakeFiles/subsys__bluetooth__controller__ll_sw__nordic.dir/lll/lll_sync_iso.c.obj
/home/jack/workspace/zephyrproject/zephyr/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c: In function ‘isr_rx’:
/home/jack/workspace/zephyrproject/zephyr/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c:833:59: warning: array subscript 1 is above array bounds of ‘uint16_t[1]’ {aka ‘short unsigned int[1]’} [-Warray-bounds=]
  833 |                         stream_handle = lll->stream_handle[stream_curr];
      |                                         ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
In file included from /home/jack/workspace/zephyrproject/zephyr/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c:32:
/home/jack/workspace/zephyrproject/zephyr/subsys/bluetooth/controller/ll_sw/nordic/../../../controller/ll_sw/lll_sync_iso.h:75:18: note: while referencing ‘stream_handle’
   75 |         uint16_t stream_handle[BT_CTLR_SYNC_ISO_STREAM_MAX];
      |                  ^~~~~~~~~~~~~
/home/jack/workspace/zephyrproject/zephyr/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync_iso.c:869:50: warning: array subscript 1 is above array bounds of ‘struct node_rx_pdu *[1][28]’ [-Warray-bounds=]
  869 |                                 if (!lll->payload[stream_curr][payload_index]) {
      |                                      ~~~~~~~~~~~~^~~~~~~~~~~~~
/home/jack/workspace/zephyrproject/zephyr/subsys/bluetooth/controller/ll_sw/nordic/../../../controller/ll_sw/lll_sync_iso.h:77:29: note: while referencing ‘payload’
   77 |         struct node_rx_pdu *payload[BT_CTLR_SYNC_ISO_STREAM_MAX]
      |                             ^~~~~~~
[196/198] Linking C executable zephyr/zephyr.elf
```

Reply:
@cvinayak does you get that when `BT_CTLR_SYNC_ISO_STREAM_MAX >= 2`? 

The whole "next BIS" part of the function should arguably be guarded by `BT_CTLR_SYNC_ISO_STREAM_MAX >= 2` right? If `BT_CTLR_SYNC_ISO_STREAM_MAX == 1` then is there any need to check more than 1 BIS?

Reply:
> does you get that when BT_CTLR_SYNC_ISO_STREAM_MAX >= 2?

yes, the native sim target compiler for some reason does not know that `lll->stream_count` would not exceed, hence, it complains about out of bounds access.

> does "Next BIS" refer to the next BIS in the BIG, and not the next BIS that we've synced to?

"Next BIS" *finds* the next BIS that we've synced to, to schedule for radio reception at the right time of the subevent.

I have removed the additional check which is not complained by the target gcc, let me see if CI complains if any.